application/forms/EventRuleConfigForm.php

```diff
@@ -415,62 +459,50 @@ protected function createRemoveButton(string $prefix): SubmitButtonElement
      *
      * @return int
      */
-    public function addOrUpdateRule(int $id, array $config): int
+    public function updateRule(int $id, array $config): int
     {
         $db = Database::get();
-
         $db->beginTransaction();
+        $db->update('rule', ['object_filter' => $config['object_filter'] ?? null], ['id = ?' => $id]);
```

Comment:
In this way, the saved value is updated, even if it remains the same if there are no changes in the filter.